spikeinterface/core/old_api_utils.py

```diff
@@ -44,36 +45,73 @@ def get_channel_property(self, channel_id, property):
         v = values[ind[0]]
         return v
 
+
 def create_extractor_from_new_recording(new_recording):
     old_recording = NewToOldRecording(new_recording)
     return old_recording
 
+
+_old_to_new_property_map = {'gain': {'name': 'gain_to_uV', 'skip_if_value': 1},
+                            'offset': {'name': 'offset_to_uV', 'skip_if_value': 0}}
+
+
 class OldToNewRecording(BaseRecording):
     """Wrapper class to convert old RecordingExtractor to a
     new Recording in spikeinterface > v0.90
-    
+
     Parameters
     ----------
     oldapi_recording_extractor : se.RecordingExtractor
         recording extractor from spikeinterface < v0.90
     """
-    
+
     def __init__(self, oldapi_recording_extractor):
         BaseRecording.__init__(self, oldapi_recording_extractor.get_sampling_frequency(),
                                oldapi_recording_extractor.get_channel_ids(),
                                oldapi_recording_extractor.get_dtype())
-        
+
         # set is_dumpable to False to use dumping mechanism of old extractor
         self.is_dumpable = False
         self.annotate(is_filtered=oldapi_recording_extractor.is_filtered)
-        
+
         # add old recording as a recording segment
         recording_segment = OldToNewRecordingSegment(oldapi_recording_extractor)
         self.add_recording_segment(recording_segment)
         self.set_channel_locations(oldapi_recording_extractor.get_channel_locations())
-        
+
+        # add old properties
+        properties = dict()
+        for ch in oldapi_recording_extractor.get_channel_ids():
+            properties_for_channel = oldapi_recording_extractor.get_channel_property_names(ch)
+            for prop in properties_for_channel:
+                prop_value = oldapi_recording_extractor.get_channel_property(ch, prop)
+                skip_if_value = None
+                if prop in _old_to_new_property_map:
+                    prop_name = _old_to_new_property_map[prop]["name"]
+                    skip_if_value = _old_to_new_property_map[prop]["skip_if_value"]
+                else:
+                    prop_name = prop
+
+                if prop_name not in properties:
+                    properties[prop_name] = dict()
+                    properties[prop_name]["ids"] = []
+                    properties[prop_name]["values"] = []
+
+                if skip_if_value is not None:
+                    if prop_value == skip_if_value:
+                        del properties[prop_name]
+                        break
+                properties[prop_name]["ids"].append(ch)
+                properties[prop_name]["values"].append(prop_value)
+
+        for property_name, prop_dict in properties.items():
+            property_ids = prop_dict["ids"]
+            property_values = prop_dict["values"]
```

Comment:
We should also change
`BaseExtractor.set_property(self, key, values, ids=None)`
to
`BaseExtractor.set_property(self, key, values, ids=None, missing=None)`
(missing=None is same behavior as before)
and then here we could handle properly and explicitly missing vales from old API.

Reply:
Let's make a separate PR for that. I don't understand exactly what you mean